packages/frontend/hooks/src/use-app-updater.ts

```diff
@@ -0,0 +1,186 @@
+import { isBrowser } from '@affine/env/constant';
+import type { UpdateMeta } from '@toeverything/infra/type';
+import { atom, useAtomValue, useSetAtom } from 'jotai';
+import { atomWithObservable, atomWithStorage } from 'jotai/utils';
+import { useCallback, useState } from 'react';
+import { Observable } from 'rxjs';
+
+function rpcToObservable<
+  T,
+  H extends () => Promise<T>,
+  E extends (callback: (t: T) => void) => () => void,
+>(
+  initialValue: T | null,
+  {
+    event,
+    handler,
+    onSubscribe,
+  }: {
+    event?: E;
+    handler?: H;
+    onSubscribe?: () => void;
+  }
+): Observable<T | null> {
+  return new Observable<T | null>(subscriber => {
+    subscriber.next(initialValue);
+    onSubscribe?.();
+    if (!isBrowser || !environment.isDesktop || !event) {
+      subscriber.complete();
+      return;
+    }
+    handler?.()
+      .then(t => {
+        subscriber.next(t);
+      })
+      .catch(err => {
+        subscriber.error(err);
+      });
+    return event(t => {
+      subscriber.next(t);
+    });
+  });
+}
+
+export const updateReadyAtom = atomWithObservable(() => {
+  return rpcToObservable(null as UpdateMeta | null, {
+    event: window.events?.updater.onUpdateReady,
+  });
+});
+
+export const updateAvailableStateAtom = atom<UpdateMeta | null>(null);
+
+export const updateAvailableAtom = atomWithObservable(get => {
+  return rpcToObservable(get(updateAvailableStateAtom), {
+    event: window.events?.updater.onUpdateAvailable,
+    onSubscribe: () => {
+      window.apis?.updater.checkForUpdatesAndNotify().catch(err => {
+        console.error(err);
+      });
+    },
+  });
+});
+
+export const downloadProgressAtom = atomWithObservable(() => {
+  return rpcToObservable(null as number | null, {
+    event: window.events?.updater.onDownloadProgress,
+  });
+});
+
+export const changelogCheckedAtom = atomWithStorage<Record<string, boolean>>(
+  'affine:client-changelog-checked',
+  {}
+);
+
+export const currentVersionAtom = atom(async () => {
+  if (!isBrowser) {
+    return null;
+  }
+  const currentVersion = await window.apis?.updater.currentVersion();
+  return currentVersion;
+});
+
+export const currentChangelogUnreadAtom = atom(async get => {
+  if (!isBrowser) {
+    return false;
+  }
+  const mapping = get(changelogCheckedAtom);
+  const currentVersion = await get(currentVersionAtom);
+  if (currentVersion) {
+    return !mapping[currentVersion];
+  }
+  return false;
+});
+
+export const isCheckingForUpdatesAtom = atom(false);
+export const isAutoDownloadUpdateAtom = atom(true);
+export const isAutoCheckUpdateAtom = atom(true);
```

Comment:
I guess these two values should be persisted like other setting values. Maybe you need to do the following when atom is mounted
- read the values from ls
- set default value in atom
- set value in electron